mule-extensions-api/src/main/java/org/mule/runtime/extension/api/describer/MuleArtifactDescriber.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.describer;
+
+import org.mule.runtime.api.meta.MuleVersion;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+
+/**
+ * Describer for a Mule artifact.
+ * <p>
+ * This object matches the mule-plugin.json element within a plugin, or any other further implementation for any Mule artifact
+ *
+ * @since 1.0
+ */
+public interface MuleArtifactDescriber {
```

Comment:
This describer is **just** the information that has being picked up from the JSON file (and the pom.xml when implemented).
There's no extra logic, such as calculating the URLs needed to feed a class loader.

The idea behind this object is to just load some bits of information that **later** each "loader" will consume to generate things like: class loader model, extension model, etc.